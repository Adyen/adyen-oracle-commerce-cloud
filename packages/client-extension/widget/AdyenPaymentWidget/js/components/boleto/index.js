import * as constants from '../../constants'
import {
    addDays,
    Checkout,
    createPresentToShopperModal,
    eventEmitter,
    createFromAction,
} from '../../utils'
import { store } from '../index'

export const setBoletoConfig = ({
    paymentMethodTypes,
    boletoDeliveryDate,
    boletoShopperStatement,
}) => {
    const hasBoleto = paymentMethodTypes.includes(
        constants.paymentMethodTypes.invoice
    )
    eventEmitter.store.emit(constants.boleto.enabled, hasBoleto)

    if (hasBoleto) {
        boletoDeliveryDate &&
            eventEmitter.store.emit(
                constants.boleto.deliveryDate,
                boletoDeliveryDate
            )
        boletoShopperStatement &&
            eventEmitter.store.emit(
                constants.boleto.shopperStatement,
                boletoShopperStatement
            )
    }
}

export const presentToShopper = customPaymentProperties => {
    const checkoutBoleto = store.get(constants.checkout.boleto)
    const options = {
        action: customPaymentProperties,
        selector: '#present-shopper',
        checkoutComponent: checkoutBoleto,
    }

    createPresentToShopperModal(() => {
        createFromAction(options)
    })
}

const createBoletoCheckout = () => {
    const checkout = new Checkout(constants.paymentMethodTypes.invoice)
    const onChange = checkout.onChange({
        deliveryDate: addDays(store.get(constants.boleto.deliveryDate)),
        shopperStatement: store.get(constants.boleto.shopperStatement),
    })

    const onSubmit = checkout.onSubmit()

    const configuration = { onSubmit, onChange }
    const order = store.get(constants.order)

    const address = order().billingAddress()

    const shopperName = {
        firstName: address.firstName(),
        lastName: address.lastName(),
    }
    const billingAddress = {
        city: address.city(),
        country: address.selectedCountry(),
        // houseNumberOrName:"952",
        postalCode: address.postalCode(),
        stateOrProvince: address.selectedState(),
        street: address.address1(),
    }
    const options = { data: { shopperName, billingAddress } }
    const checkoutOptions = {
        configuration,
        selector: '#adyen-boleto-payment',
        type: 'boletobancario',
        options,
    }

    checkout.createCheckout(checkoutOptions, checkout => {
        eventEmitter.store.emit(constants.checkout.boleto, checkout)
    })
}

export default createBoletoCheckout
