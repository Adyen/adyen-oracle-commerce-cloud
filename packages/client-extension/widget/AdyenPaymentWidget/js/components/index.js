export { default as Component } from './component'
export { default as Order } from './order'
export { default as Payment } from './payment'
export { default as store } from './store'
export {
    default as createBoletoCheckout,
    presentToShopper,
    setBoletoConfig,
} from './boleto'
export { default as createCardCheckout } from './card'
