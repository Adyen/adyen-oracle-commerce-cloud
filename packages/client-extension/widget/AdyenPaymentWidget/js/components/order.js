import $ from 'jquery'
import ccRestClient from 'ccRestClient'
import ccConstants from 'ccConstants'
import storageApi from 'storageApi'
import pubsub from 'pubsub'
import {
    redirectAuth,
    createSpinner,
    getOrderPayload,
    eventEmitter,
} from '../utils'
import * as constants from '../constants'
import { store } from './index'

class Order {
    constructor() {
        this.instance = storageApi.getInstance()

        this.startEventListeners()
    }

    startEventListeners = () => {
        eventEmitter.order.on(
            constants.pageChanged,
            this.getUrlParametersAndCreateOrder
        )
        eventEmitter.order.on(
            constants.initialOrderCreated,
            this.initialOrderCreated
        )
    }

    initialOrderCreated = orderEvent => {
        const order = store.get(constants.order)
        const {
            customPaymentProperties: { data, resultCode },
        } = orderEvent.order.payments[0]

        const payload = {
            order: getOrderPayload(order),
            customPaymentProperties: JSON.parse(data),
            resultCode,
        }

        eventEmitter.store.emit(constants.orderPayload, payload)
        const isDone = store.get(constants.isDone)
        !isDone() && redirectAuth(payload, this.createOrder)
    }

    getAndCreatOrder = result => {
        result.paymentData = this.instance.getItem(
            constants.storage.paymentData
        )

        const payment = { type: 'generic', customProperties: result }

        this.recreateOrder(payment)
    }

    createOrder = orderPayload => {
        const translate = store.get(constants.translate)
        const orderFail = ({ message = 'Failed to create order' } = {}) => {
            $.Topic(pubsub.topicNames.ORDER_SUBMISSION_FAIL).publish({
                message: 'fail',
                errorMessage: message,
            })
        }

        const submitOrder = ({ id, uuid }) => {
            const publishData = { message: 'success', id, uuid }
            $.Topic(pubsub.topicNames.ORDER_COMPLETED).publish(publishData)
            $.Topic(pubsub.topicNames.ORDER_SUBMISSION_SUCCESS).publish([
                publishData,
            ])
        }

        const orderSuccess = data => {
            const { paymentState } = data.payments[0]
            const isAuthorized =
                paymentState === ccConstants.PAYMENT_GROUP_STATE_AUTHORIZED
            const message = translate(
                constants.errorMessages.failed3dsValidation
            )
            isAuthorized ? submitOrder(data) : orderFail({ message })
        }

        ccRestClient.request(
            ccConstants.ENDPOINT_ORDERS_CREATE_ORDER,
            orderPayload,
            orderSuccess,
            orderFail
        )
    }

    recreateOrder = paymentData => {
        const storedOrder = JSON.parse(
            this.instance.getItem(constants.storage.order)
        )
        storedOrder.payments = [paymentData]

        eventEmitter.store.emit(constants.isDone, true)
        this.createOrder(storedOrder)
    }

    getResult = urlParameters => {
        const params = urlParameters.split('&')
        const formatResult = (acc, param) => {
            const [key, value] = param.split('=')
            return { ...acc, [key]: decodeURIComponent(value) }
        }

        return params.reduce(formatResult, {})
    }

    getUrlParametersAndCreateOrder = pageData => {
        const order = store.get(constants.order)
        const { parameters } = pageData

        if (parameters) {
            createSpinner()

            const parametersObj = this.getResult(parameters)

            if (order().paymentGateway()) {
                order().paymentGateway().type = ''
            }

            this.isAction = 'PaRes' in parametersObj && 'MD' in parametersObj
            this.isAction && this.getAndCreatOrder(parametersObj)
        }
    }
}

export default Order
