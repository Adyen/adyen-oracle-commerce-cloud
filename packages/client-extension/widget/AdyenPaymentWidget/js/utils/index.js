export { createSpinner, destroySpinner } from './spinner'
export { default as ajax } from './ajax'
export { default as redirectAuth } from './redirectAuth'
export { default as getOrderPayload } from './getOrderPayload'
export { default as eventEmitter } from './events'
export { default as getInstallmentOptions } from './getInstallmentOptions'
export { default as Checkout, createFromAction } from './checkout'
export { createPresentToShopperModal } from './modal'
export { default as createError } from './createError'
export { addDays } from './date'
