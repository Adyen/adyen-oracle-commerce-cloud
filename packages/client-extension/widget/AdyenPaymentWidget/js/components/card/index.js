import { store } from '../index'
import * as constants from '../../constants'
import { Checkout, eventEmitter, getInstallmentOptions } from '../../utils'

export const onBrand = ({ brand }) => {
    eventEmitter.store.emit(constants.selectedBrand, brand)
    const installmentsOptions = store.get(constants.installmentsOptions)
    const cart = store.get(constants.cart)
    const amount = cart().amount()
    const installments = installmentsOptions ? getInstallmentOptions(installmentsOptions, amount, brand) : []

    eventEmitter.store.emit(constants.installments, installments)
}

export const onConfigSuccess = () => eventEmitter.store.emit(constants.isLoaded, true)

const createCardCheckout = () => {
    // const [card] = paymentMethods.filter(paymentMethod => paymentMethod.type === constants.paymentMethodTypes.scheme)

    // if (card) {
    //     eventEmitter.store.emit(constants.paymentMethodTypes.scheme, true)
    //     const { brands } = card
    const checkout = new Checkout(constants.paymentMethodTypes.scheme)
    const onSubmit = checkout.createOnSubmit()
    const onChange = checkout.createOnChange()
    const cardComponent = document.querySelector('adyen-payment-method-card')
    cardComponent.addEventListener('adyenChange', onChange)
    cardComponent.addEventListener('adyenSubmit', onSubmit)

    eventEmitter.store.emit(constants.checkout.card, checkout)
    // const cardComponent = document.querySelector('adyen-payment-method-card')

    //     const onChange = checkout.onChange()
    //     const onSubmit = checkout.onSubmit()

        // const storedPaymentType = store.get(constants.storedPaymentType)
        // const options = { hasHolderName: false, enableStoreDetails: !!storedPaymentType(), brands }

        // const configuration = { onBrand, onConfigSuccess, onChange, onSubmit }
        // const checkoutOptions = { configuration, selector: '#adyen-card-payment', type: 'card', options }

        // checkout.createCheckout(checkoutOptions, checkout => {
        //     eventEmitter.store.emit(constants.checkout.card, checkout)
        // })
    // }
}

export default createCardCheckout
