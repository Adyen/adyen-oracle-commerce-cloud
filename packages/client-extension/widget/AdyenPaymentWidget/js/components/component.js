import * as constants from '../constants'
import { eventEmitter } from '../utils'
import { createBoletoCheckout, createCardCheckout, store } from './index'

class Component {
    constructor() {
        this.startEventListeners()
    }

    startEventListeners = () => {
        eventEmitter.component.on(constants.render, this.render)
        eventEmitter.component.on(
            constants.comboCardOptions,
            this.getComboCardOptions
        )
    }

    getComboCardOptions = () => {
        const brands = { visa: constants.bins.electron }
        const selectedComboCard = store.get(constants.selectedComboCard)()
        const isDebitCard = selectedComboCard === constants.comboCards.debit
        const brand = store.get(constants.selectedBrand)
        const selectedBrand =
            brand in brands
                ? brands[store.get(constants.selectedBrand)]
                : store.get(constants.selectedBrand)

        const options = {
            selectedBrand,
            additionalData: { overwriteBrand: true },
        }

        eventEmitter.store.emit(
            constants.comboCardOptions,
            isDebitCard ? options : {}
        )
    }

    render = () => {
        store.get(constants.ajax)('paymentMethods', this.getPaymentMethods)
    }

    getPaymentMethods = paymentMethodsResponse => {
        eventEmitter.store.emit(
            constants.paymentMethodsResponse,
            paymentMethodsResponse
        )

        createCardCheckout()
        store.get(constants.brazilEnabled) && createBoletoCheckout()
    }
}

export default Component
