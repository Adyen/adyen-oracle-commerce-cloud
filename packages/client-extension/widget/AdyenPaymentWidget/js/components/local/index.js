import * as constants from '../../constants'
import { Checkout, eventEmitter } from '../../utils'

const hasType = paymentMethod => 'type' in paymentMethod
const isNotScheme = paymentMethod => hasType(paymentMethod) && paymentMethod.type !== 'scheme'
const hasDetails = paymentMethod => isNotScheme(paymentMethod) && 'details' in paymentMethod
const hasOneItem = paymentMethod => hasDetails(paymentMethod) && paymentMethod.details.length === 1
const getDetail = paymentMethod => hasOneItem(paymentMethod) && paymentMethod.details[0]
const isValid = (paymentMethod, key, value) => getDetail(paymentMethod) && getDetail(paymentMethod)[key] === value

const isLocalPaymentMethod = paymentMethod => {
    const isIssuer = isValid(paymentMethod, 'key', 'issuer')
    const isSelect = isValid(paymentMethod, 'type', 'select')
    const isLocal = isIssuer && isSelect

    return isLocal
}

const createLocalComponent = ({ type, name }) => {
    const headingNode = document.createElement('div')
    headingNode.classList.add('collapse-heading')
    headingNode.innerHTML = ` <strong>
        <span
            class="collapse-toggle collapsed"
            data-toggle="collapse"
            data-target="#adyen-${type}-wrapper"
        >
          <button>${name}</button>
        </span>
      </strong>`
    const component = document.createElement('div')
    component.setAttribute('id', `adyen-${type}-wrapper`)
    component.classList.add('adyen-wrapper', 'collapse')

    component.innerHTML = `
        <div class="loader-wrapper hide">
            <div class="lds-ring"><div></div><div></div><div></div><div></div></div>
            <div style="font-weight: bold; font-size: 16px;">Processing...</div>
        </div>
        <div id="adyen-${type}-payment" class="adyen-component"></div>
    `
    const adyenNode = document.getElementById('adyen-collapse-heading')
    adyenNode.appendChild(headingNode)
    adyenNode.appendChild(component)
}

const createLocalCheckout = (acc, localPaymentMethod) => {
    const { type } = localPaymentMethod
    const checkout = new Checkout(type)
    const onChange = checkout.onChange()
    const onSubmit = checkout.onSubmit()

    const configuration = { onSubmit, onChange, showPayButton: true }

    createLocalComponent(localPaymentMethod)
    const checkoutOptions = type => ({ configuration, selector: `#adyen-${type}-payment`, type })
    checkout.createCheckout(checkoutOptions(type), checkout => ({ ...acc, [type]: checkout }))
}

const createLocalPaymentCheckout = ({ paymentMethods }) => {
    const localPaymentMethods = paymentMethods.filter(isLocalPaymentMethod)
    const localPaymentMethodsCheckout = localPaymentMethods.reduce(createLocalCheckout, {})

    eventEmitter.store.emit(constants.checkout.local, localPaymentMethodsCheckout)
}

export default createLocalPaymentCheckout
