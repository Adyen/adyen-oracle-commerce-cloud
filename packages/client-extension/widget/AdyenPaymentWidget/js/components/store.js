import ko from 'knockout'
import { comboCards, noInstallmentsMsg, countries } from '../constants'
import { ajax, eventEmitter } from '../utils'
import { Component, Order, Payment } from './index'

class Store {
    constructor() {
        this.startEventListeners()
    }

    brazilEnabled = false

    storedPaymentType = ko.observable('')
    environment = undefined
    pageParams = undefined

    // Component
    component = new Component()
    checkout = undefined
    originKey = ''
    isValid = false
    selectedBrand = undefined
    creditCard = comboCards.credit
    debitCard = comboCards.debit
    comboCards = ko.observable([comboCards.credit, comboCards.debit])
    comboCardOptions = {}
    selectedComboCard = ko.observable(comboCards.credit)
    isPaymentStored = ko.observable(false)
    paymentMethodsResponse = undefined
    paymentDetails = undefined
    isLoaded = ko.observable(false)
    installments = ko.observable([])
    selectedInstallment = ko.observable({})
    installmentsAllowedCountries = [countries.mexico.locale, countries.brazil.locale]
    installmentsOptions = []

    // Payment
    paymentComponent = new Payment()
    genericPayment = { type: 'generic' }

    // Order
    orderComponent = new Order()
    isDone = ko.observable(false)

    // Boleto
    boletoEnabled = false
    boletoDeliveryDate = ''
    boletoShopperStatement = ''

    // Local
    checkoutLocal = ko.observable([])
    localPaymentMethods = ko.observableArray()

    get installmentsEnabled() {
        const hasInstallments = !!this.installments().length
        const cardIsCredit = this.selectedComboCard() === comboCards.credit
        return ko.observable(this.isAllowedCountryForInstallments && hasInstallments && cardIsCredit)
    }

    get isAllowedCountryForInstallments() {
        return this.installmentsAllowedCountries.includes(this.locale.toLowerCase())
    }

    startEventListeners = () => {
        eventEmitter.store.listen((key, value) => {
            if (typeof this[key] === 'function') {
                this[key](value)
            } else {
                this[key] = value
            }
        })
    }

    get = key => (this.has(key) ? this[key] : undefined)

    has = key => key in this

    init = viewModel => {
        this.id = viewModel.id
        this.order = viewModel.order
        this.cart = viewModel.cart
        this.ajax = ajax(viewModel.isPreview())
        this.translate = viewModel.translate
        this.locale = viewModel.locale()
        this.noInstallmentsMsg = viewModel.translate(noInstallmentsMsg)
    }
}

export default new Store()
