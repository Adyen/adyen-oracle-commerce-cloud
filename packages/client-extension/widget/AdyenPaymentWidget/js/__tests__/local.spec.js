import Widget from '../../../../__mocks__/widget'
import paymentMethodsResponseMock from '../../../../__mocks__/paymentMethods'
import { createLocalPaymentCheckout } from '../components'
import store from '../components/store'

jest.mock('../utils/checkout')
import { Checkout, eventEmitter } from '../utils'
import generateTemplate, { mockTemplate } from '../utils/tests/koTemplate'
import { isLocalPaymentMethod, submitPayByLink } from '../components/local'
import * as constants from '../constants'

describe('Local', () => {
    let widget
    let tmplWidget
    let template

    beforeEach(() => {
        jest.clearAllMocks()
        tmplWidget = mockTemplate('Tmpl_Widget')
        widget = new Widget()
        template = generateTemplate(widget)

        Checkout.prototype.createCheckout = jest.fn()
        Checkout.prototype.onChange = jest.fn()
        Checkout.prototype.onSubmit = jest.fn()
    })

    it('should create local checkout component', function() {
        createLocalPaymentCheckout(paymentMethodsResponseMock)
        expect(Checkout.prototype.createCheckout).toHaveBeenCalled()
    })

    it('should have 5 items', function() {
        createLocalPaymentCheckout(paymentMethodsResponseMock)
        expect(Checkout.prototype.createCheckout.mock.calls).toHaveLength(16)
    })

    it('should have correct options', function() {
        createLocalPaymentCheckout(paymentMethodsResponseMock)
        const { type, selector } = Checkout.prototype.createCheckout.mock.calls[0][0]
        const expectedType = 'directEbanking'
        expect(type).toEqual(expectedType)
        expect(selector).toEqual(`#adyen-${expectedType}-payment`)

        const el = document.querySelector(`#adyen-${expectedType}-payment`)
        expect(el).not.toBeUndefined()
    })

    it('should have correct callback', function() {
        createLocalPaymentCheckout(paymentMethodsResponseMock)
        const cb = Checkout.prototype.createCheckout.mock.calls[0][1]
        expect('directEbanking' in cb()).toBeTruthy()
    })

    it('should render', function() {
        createLocalPaymentCheckout(paymentMethodsResponseMock)
        expect(template).toMatchSnapshot()
    })

    it('should create details', function() {
        const [localPaymentMethod] = paymentMethodsResponseMock.paymentMethods.filter(isLocalPaymentMethod)
        submitPayByLink(localPaymentMethod)
        const { type, name, supportsRecurring } = localPaymentMethod
        const paymentDetails = store.get(constants.paymentDetails)
        const { shippingAddress } = store.get(constants.order)()
        const countryCode = shippingAddress().selectedCountry()
        const paymentMethod = { countryCode, name, supportsRecurring, type }
        const expected = { [type]: { paymentMethod } }
        expect(paymentDetails).toEqual(expected)
    })
})
