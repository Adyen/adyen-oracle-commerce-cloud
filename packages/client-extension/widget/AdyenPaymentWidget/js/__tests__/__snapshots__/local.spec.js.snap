// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local should render 1`] = `
Object {
  Symbol(dom): <body>
    <link
      href="https://checkoutshopper-live.adyen.com/checkoutshopper/sdk/3.3.0/adyen.css"
      rel="stylesheet"
    />
    <legend />
    <div
      class="oc-panel panel-1"
      data-oc-id="panel-1"
      id="adyen-collapse-heading"
    >
      <div
        class="collapse-heading"
      >
        <strong>
           
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-generic-wrapper"
            data-toggle="collapse"
          >
            <button />
          </span>
        </strong>
      </div>
      <div
        class="adyen-wrapper collapse"
        id="adyen-generic-wrapper"
      >
        <div
          class="loader-wrapper hide"
        >
          <div
            class="lds-ring"
          >
            <div />
            <div />
            <div />
            <div />
          </div>
          <div
            style="font-weight: bold; font-size: 16px;"
          >
            Processing...
          </div>
        </div>
        <div
          class="adyen-component"
          id="adyen-card-payment"
        />
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-directEbanking-wrapper"
            data-toggle="collapse"
          >
            <button>
              Online bank transfer.
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-directEbanking-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-directEbanking-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-dotpay-wrapper"
            data-toggle="collapse"
          >
            <button>
              Local Polish Payment Methods
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-dotpay-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-dotpay-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-entercash-wrapper"
            data-toggle="collapse"
          >
            <button>
              Bank Payment
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-entercash-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-entercash-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-eps-wrapper"
            data-toggle="collapse"
          >
            <button>
              EPS
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-eps-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-eps-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-ideal-wrapper"
            data-toggle="collapse"
          >
            <button>
              iDEAL
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-ideal-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-ideal-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-interac-wrapper"
            data-toggle="collapse"
          >
            <button>
              InteracÂ® Online
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-interac-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-interac-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-klarna-wrapper"
            data-toggle="collapse"
          >
            <button>
              Pay later with Klarna.
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-klarna-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-klarna-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-klarna_account-wrapper"
            data-toggle="collapse"
          >
            <button>
              Slice it with Klarna.
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-klarna_account-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-klarna_account-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-klarna_paynow-wrapper"
            data-toggle="collapse"
          >
            <button>
              Pay now with Klarna.
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-klarna_paynow-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-klarna_paynow-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-molpay_cash-wrapper"
            data-toggle="collapse"
          >
            <button>
              7-Eleven
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-molpay_cash-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-molpay_cash-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-molpay_epay-wrapper"
            data-toggle="collapse"
          >
            <button>
              epay
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-molpay_epay-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-molpay_epay-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-multibanco-wrapper"
            data-toggle="collapse"
          >
            <button>
              Multibanco
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-multibanco-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-multibanco-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-openbanking_UK-wrapper"
            data-toggle="collapse"
          >
            <button>
              Online banking
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-openbanking_UK-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-openbanking_UK-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-paysafecard-wrapper"
            data-toggle="collapse"
          >
            <button>
              Paysafecard
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-paysafecard-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-paysafecard-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-unionpay-wrapper"
            data-toggle="collapse"
          >
            <button>
              UnionPay
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-unionpay-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-unionpay-payment"
          />
        </div>
      </div>
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-wechatpayQR-wrapper"
            data-toggle="collapse"
          >
            <button>
              WeChat Pay
            </button>
          </span>
        </strong>
        <div
          class="adyen-wrapper collapse"
          id="adyen-wechatpayQR-wrapper"
        >
          <div
            class="loader-wrapper hide"
          >
            <div
              class="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <div
              style="font-weight: bold; font-size: 16px;"
            >
              Processing...
            </div>
          </div>
          <div
            class="adyen-component"
            id="adyen-wechatpayQR-payment"
          />
        </div>
      </div>
    </div>
  </body>,
}
`;
